backend/src/database/services/database-service.ts

```diff
@@ -1,29 +1,106 @@
+import { uuid_ossp } from '@electric-sql/pglite/contrib/uuid_ossp';
+import * as schema from '../schema.js';
+import { Config } from '../../config.js';
+
+export type DatabaseConnectionMode = 'baseline' | 'default' | 'testing';
+
+export let testingDatabaseName: string;
 
 export class DatabaseService {
-    public readonly actionWrapperService: ActionWrapperService;
-    public readonly exerciseWrapperService: ExerciseWrapperService;
-
-    public constructor(private readonly dataSource: DataSource) {
-        this.exerciseWrapperService = new ExerciseWrapperService(dataSource);
-        this.actionWrapperService = new ActionWrapperService(
-            this.exerciseWrapperService,
-            dataSource
-        );
+    public readonly databaseConnection: DatabaseConnection;
+    private _initialized = false;
+
+    public constructor(dbConnection: DatabaseConnection) {
+        this.databaseConnection = dbConnection;
+        this._initialized = true;
     }
 
     public get isInitialized() {
-        return this.dataSource.isInitialized;
+        return this._initialized;
     }
 
-    /**
-     * Wrap all operations into one database transaction. Note that all calls inside MUST use the provided manager.
-     */
-    public async transaction<T>(
-        operation: (manager: EntityManager) => Promise<T>
-    ) {
-        return this.dataSource.transaction(operation);
+    public static async createNewDatabaseConnection(
+        mode: DatabaseConnectionMode = 'default'
+    ): Promise<DatabaseService> {
+        Config.initialize();
+        if (Config.useDb) {
+            const defaultDatabaseName = `${Config.dbName}`;
+            testingDatabaseName = `${Config.dbName}_TESTING`;
+            const connection = postgresDrizzle({
+                connection: {
+                    host: Config.dbHost,
+                    port: Config.dbPort,
+                    database:
+                        mode === 'baseline'
+                            ? // This database probably always exists
+                              'postgres'
+                            : mode === 'default'
+                              ? defaultDatabaseName
+                              : testingDatabaseName,
+                    user: Config.dbUser,
+                    password: Config.dbPassword,
+                    ssl: false,
+                },
+                logger: Config.dbLogging,
+                schema,
+            });
+            await this.testConnection(connection);
+            return new DatabaseService(connection);
+        }
+
+        const pgLite = new PGlite({
+            extensions: { uuid_ossp },
+        });
+        const db = pgliteDrizzle({
+            client: pgLite,
+            schema,
+            logger: Config.dbLogging,
+        });
+        await this.testConnection(db);
+        await db.execute(`CREATE EXTENSION IF NOT EXISTS "uuid-ossp";`);
+        await this.migrate(db);
+        return new DatabaseService(db);
```

Comment:
I think I would split this into two private methods `createNewPostgresDatabaseConnection` and `createNewPgliteDatabaseConnection` and use an explicit if/else (similar to `migrate`).